### PR TITLE
feat(RadioButton): migrate to new design

### PR DESCRIPTION
**Pull request by eszthoff**
Story: [ONEUI-445](https://textkernel.atlassian.net/browse/ONEUI-445)

### CODE REVIEW

File: src/components/RadioButton/RadioButton.scss [hunk 1]
```diff
@@ -1,102 +1,123 @@
...
-            .RadioButton__label {
-                .RadioButton__box {
-                    border-color: var(--color-neutral-70);
-                }
-            }
+        /* stylelint-disable no-descending-specificity */
+        .RadioButton__label .RadioButton__box {
+         /* stylelint-enable no-descending-specificity */
+         border-color: var(--color-border-disabled, #CCC);
+            background: var(--color-background-input-default);
```
Comment (mukiienko) [inline src/components/RadioButton/RadioButton.scss:82]: Please align spaces.

File: src/components/RadioButton/RadioButton.scss [hunk 2]
```diff
@@ -1,102 +1,123 @@
...
 
     &--disabled {
-        &:hover {
-            .RadioButton__label {
-                .RadioButton__box {
-                    border-color: var(--color-neutral-70);
-                }
-            }
+        /* stylelint-disable no-descending-specificity */
+        .RadioButton__label .RadioButton__box {
```
Comment (mukiienko) [inline src/components/RadioButton/RadioButton.scss:79]: Can we define this style without `.RadioButton__label` part and then remove `/* stylelint-disable no-descending-specificity */`?

File: src/components/RadioButton/RadioButton.scss [hunk 3]
```diff
@@ -1,102 +1,123 @@
...
+         /* stylelint-enable no-descending-specificity */
+         border-color: var(--color-border-disabled, #CCC);
+            background: var(--color-background-input-default);
         }
 
-        .RadioButton__input{
+        .RadioButton__input {
+            /* stylelint-disable no-descending-specificity */
             + .RadioButton__label {
```
Comment (mukiienko) [inline src/components/RadioButton/RadioButton.scss:87]: Can we define `RadioButton__label` style without `.RadioButton__input` part and rid of /* stylelint-disable no-descending-specificity */?

**Comment by mukiienko** [comment]
I noticed a strange animation after selecting the radio button: first, it fills with one blue circle, and then another blue circle is added inside. Is this intended? 

https://github.com/user-attachments/assets/b42db47c-0302-4379-a971-57c8c36f9c0a

File: src/components/RadioButton/RadioButton.scss [hunk 4]
```diff
@@ -1,102 +1,123 @@
...
+        }
     }
 
     &__input {
-        display: none;
+        margin: 0;
+        width: 0;
+        height: 0;
+        border: none;
+        appearance: none;
```
[the remark this answers is not in the document]
Comment (eszthoff) [reply src/components/RadioButton/RadioButton.scss:39]: Thanks. There was one more thing I learned thanks to these last PRS. I posted them in #frontend 

Where do you think being visible, that these 2 are still needed?

File: src/components/RadioButton/RadioButton.scss (hunk 2 above)
Comment (eszthoff) [reply src/components/RadioButton/RadioButton.scss:79]: no, even if you remove it, the issue is still there. I tried for awhile to move things around, but it creates more problem then it solves

File: src/components/RadioButton/RadioButton.scss (hunk 3 above)
Comment (eszthoff) [reply src/components/RadioButton/RadioButton.scss:87]: no, I was playing around with it quite a lot, but it is the least things that go funny (need to be ignored)

**Comment by eszthoff** [comment]
> I noticed a strange animation after selecting the radio button: first, it fills with one blue circle, and then another blue circle is added inside. Is this intended?
> 
>  Screen.Recording.2024-07-25.at.11.36.09.mov

Nice catch. I made it smoother. It is still 2 steps, but it is not so jumpy. I hope it is ok like this (it is not in the designs, it was a leftover from the previous implementation)

File: src/components/RadioButton/RadioButton.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/components/RadioButton/RadioButton.scss:39]: For this use case I believe your approach for hiding elements is sufficient.

File: src/components/RadioButton/RadioButton.scss (hunk 2 above)
Comment (mukiienko) [reply src/components/RadioButton/RadioButton.scss:79]: ok, let's keep as it's now.

**Comment by mukiienko** [comment]
> > I noticed a strange animation after selecting the radio button: first, it fills with one blue circle, and then another blue circle is added inside. Is this intended?
> > Screen.Recording.2024-07-25.at.11.36.09.mov
> 
> Nice catch. I made it smoother. It is still 2 steps, but it is not so jumpy. I hope it is ok like this (it is not in the designs, it was a leftover from the previous implementation)

For me it still doesn't look smooth enough, but let's leave this point for the PO to decide.

**Pull request merged**